packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -155,10 +166,33 @@ function BlockPatternList(
 	},
 	ref
 ) {
-	const composite = useCompositeState( { orientation } );
+	const compositeStore = useCompositeStore( {
+		orientation,
+		setItems: ( items ) => {
+			// This is necessary for if/when we filter the block patterns;
+			// we can potentially remove the currently active item, so we
+			// check to see if the active item is still present, and if not,
+			// reset the active item to be the first available block.
+
+			const currentIds = items.map( ( item ) => item.id );
+			if ( ! currentIds.includes( activeId ) ) {
+				// We can't rely on the order of `currentIds` here, because
+				// `blockPatterns` may not be in the same order the blocks were
+				// originally registered in. So we filter the blocks by what's
+				// visible, and take the first item in that list instead.
+				const firstPattern = blockPatterns.filter( ( pattern ) =>
+					currentIds.includes( pattern.name )
+				)[ 0 ];
+				compositeStore.setActiveId( firstPattern?.name );
+			}
+		},
+	} );
```

Comment:
Similarly to [this comment](https://github.com/WordPress/gutenberg/pull/55344#discussion_r1397037149), I would have assumed that ariakit would be able to automatically update the selection.

Also, in case we still needed to manually update the selection, I wonder if using a `useEffect` (or `useLayoutEffect`) would be a better choice here:

```
const items = compositeStore.useState( 'items' );

useEffect(() => {
  // logic
}, [ items, activeId ] };
```

Reply:
(cc @diegohaz )

Reply:
In this case, I'd suggest resetting the active ID to undefined each time the list is filtered. This could likely be an effect tied to the `blockPatterns` prop as a dependency.

However, if you're interested in listening to items in their correct order, the `renderedItems` state can be used:

```js
const items = compositeStore.useState( 'renderedItems' );

useEffect( () => {
  // logic
}, [ items, activeId ] );
```

As I've previously stated, unlike Reakit, Ariakit does not automatically update `activeId` because there are valid reasons to maintain it, even when the element is removed from the DOM.

Looking ahead, we plan to introduce a separate feature to address this. It might look something like this:

```jsx
<CompositeItem render={ <CompositeDisposable /> }>
```